lib/px/utils/pxclient.lua

```diff
@@ -160,6 +165,74 @@ function M.load(config_file)
         _M.submit(cjson.encode(enforcer_telemetry), px_constants.TELEMETRY_PATH);
     end
 
+    function _M.forward_to_perimeterx(server, port_overide)
+        ngx_req_set_header(px_constants.ENFORCER_TRUE_IP_HEADER, px_headers.get_ip())
+        local port = ngx.var.scheme == 'http' and '80' or '443'
+        if port_overide ~= nil then
+            px_logger.debug('Overrding port ' .. port ..  ' => ' .. port_overide)
+            port =  port_overide
+        end
+        px_logger.debug("Using " .. ngx.var.scheme .. " port " .. port)
+        local httpc = http.new()
+
+        httpc:set_timeout(500)
+
+        local ok, err = httpc:connect(server, port)
+
+        if not ok then
+            ngx.log(ngx.ERR, err)
+            return
+        end
+
+        if port == '443' and ssl_enabled then
+            if ssl_enabled == true then
+                local session, err = httpc:ssl_handshake()
+                if not session then
+                    px_logger.error("HTTPC SSL handshare error: " .. err)
+                end
+            end
+        end
+
+        httpc:set_timeout(2000)
+        httpc:proxy_response(httpc:proxy_request())
+        httpc:set_keepalive()
+    end
+
+    function _M.reverse_px_client()
+        local px_request_uri = "/" .. px_config.px_appId .. "/main.min.js"
```

Comment:
shouldn't this be the appId without the 'PX' prefix?

Reply:
no, this is where to redirect the requset to